addon/serializers/web-api.js

```diff
@@ -45,22 +48,32 @@ export default DS.RESTSerializer.extend({
     }
   },
 
-  extractErrors: function(store, typeClass, payload, id) {
-    let modelState = payload.modelState,
-        payloadKey = `${typeClass.modelName}.`,
-        formattedPayload = { errors: { } },
-        key;
+  normalizeErrors: function normalizeErrors(typeClass, payload) {
+    let payloadKey = `${typeClass.modelName}.`,
+        keys = Object.keys(payload);
 
-    for(key in modelState) {
-      if(modelState.hasOwnProperty(key)) {
-        formattedPayload.errors[key.replace(payloadKey, '').camelize()] = modelState[key];
+    keys.forEach(function(key) {
+      if(payload.hasOwnProperty(key)) {
+        payload[key.replace(payloadKey, '').camelize()] = payload[key];
+        delete payload[key];
       }
-    }
-
-    return this._super(store, typeClass, formattedPayload, id);
+    });
+    return this._super(typeClass, payload);
+  },
+  clearModelName: function(errors, modelName) {
```

Comment:
https://github.com/emberjs/data/blob/v1.13.8/packages/ember-data/lib/serializers/json-serializer.js#L1578

But they changed it again.. I'll create a new PR for the latest stable release.